tests/components/modbus/test_init.py

```diff
@@ -53,7 +53,6 @@
     CONF_HVAC_MODE_COOL,
     CONF_HVAC_MODE_DRY,
     CONF_HVAC_MODE_HEAT,
-    CONF_HVAC_MODE_HEAT_COOL,
```

Comment:
Wh6 remove test of this mode, that does not seem correct.

Reply:
The mode is still used, see line 270 in __init__.py

Apart from that you cannot just remove a configuration parameter, it first goes deprecated and after 6 month is can be removed.

Reply:
It is also in use in several tests:

```
test_climate.py:    CONF_HVAC_MODE_HEAT_COOL,
test_climate.py:                            CONF_HVAC_MODE_HEAT_COOL: 3,
test_init.py:    CONF_HVAC_MODE_HEAT_COOL,
test_init.py:                        CONF_HVAC_MODE_HEAT_COOL: 2,
test_init.py:                        CONF_HVAC_MODE_HEAT_COOL: 2,
test_init.py:                        CONF_HVAC_MODE_HEAT_COOL: 2,
```

Your change means that configurations with CONF_HVAC_MODE_HEAT_COOL are not tested, so not sure why you think it does not decrease test.

Reply:
Maybe there is a misunderstanding. 
I replaced the old testing code and validation routines written when I've added the support for multiple target_temp registers and so those modes are not used anymore **in the tests**. Previously we are not testing that specific mode, we are only using its value for testing the conflics. Now that routine has been replaced and that import is not used anymore inside that file, but conflicts are still tested, but in another way.

Reply:
Please do not modify tests in a PR that contains production code ! the reason being we use the tests to see if the production code have side effects.

When tests are modified with production code, it's because they no longer work, and thus we have to review even more carefully.

Apart from that it is important to test every possible parameter in the tests, and a test set that do not do that will not b3 accepted.

Reply:
In essence your change indicated that HEAT_COOL no longer works, and I assume that is not the case.

Looking at your new code, I cannot see that we get an error if HEAT_COOL is accidentally removed from the production code.

Reply:
@janiversen  Ok, I've understood the side effect you tell me. I've added again the HEAT_COOL mode into a config test.